Assets/MixedRealityToolkit/Interfaces/SpatialAwareness/IMixedRealitySpatialAwarenessSystem.cs

```diff
@@ -45,7 +45,6 @@ public interface IMixedRealitySpatialAwarenessSystem : IMixedRealityEventSystem
         /// <returns>
         /// Read only copy of the list of registered observers.
         /// </returns>
-        [Obsolete("GetObservers will be removed in a future release. Check to see if the instance implements IMixedRealityDataProviderAccess and call GetDataProviders.")]
```

Comment:
The obsolete attribute should not be removed. you need to cast the spatial awareness instance to IMixedRealityDataProviderAccess to call GetProviders().

Reply:
(SpatialAwarenessSystem as IMixedRealityDataProviderAccess)?.GetProviders()

Reply:
i am happy to work with you on the documentation, etc :)

Reply:
#4737 fixes the error in spatial awareness where it was not specifying that it implements IMixedRealityDataProviderAccess

Reply:
At the time when I wrote this, IMixedRealityDataProviderAccess didn't exist on this object. We probably shouldn't have obsoleted these functions unless the alternative actually existed.

Reply:
What I'll do is merge the stabilization branch down to dev and then get this change updated

Reply:
Okay, merged in. I have to say that now that I've had some experience playing with this, the new API design isn't intuitive and doesn't really make sense to me - it feels like this is designed to be extendable and generic but errs so far on that side that usabillity suffers. We can probably chat about this on an issue afterward and isn't worth going into on this PR.

Reply:
To be clear, the new API surface with respect to how it applies in the SA system. I don't have an opinion yet on the others.

Reply:
The reason for the new API design was to reduce breaking changes and add flexibility for implementations to add providers in the future.

We can go into more detail if desired :)